gensim/utils.py

```diff
@@ -1263,3 +1269,18 @@ def _iter_windows(document, window_size, copy=False, ignore_below_size=True):
     else:
         for doc_window in doc_windows:
             yield doc_window.copy() if copy else doc_window
+
+
+def walk_with_depth(top, topdown=True, onerror=None, followlinks=False):
+    """Wrap `os.walk` in code that analyzes the directory path of each yield to
+    determine the depth relative to `top`. Yields in the `top` directory are at
+    depth 0.
+
+    Returns:
+        generator of tuples of (depth, dirpath, dirnames, filenames).
+    """
+    path = os.path.abspath(top)
+    for dirpath, dirnames, filenames in os.walk(path, topdown, onerror, followlinks):
+        sub_path = dirpath.replace(path, '')
+        depth = sub_path.count(os.sep)
```

Comment:
This construct still makes me a little uneasy. Can we at least `os.path.normpath`, to get rid of any double/trailing/leading slashes? Or does `os.walk` normalize the `dirpath` somehow? Although in that case, we'd have to normalize `path` and `dirpath` in exactly the same way, so that the `.replace()` above works.